### PR TITLE
add hint to options to help user

### PR DESCRIPTION
**Pull request by dragon512**
Ran cmake-format

### CODE REVIEW

**Review by JosiahWI — APPROVED** [review]
Very nice addition to improve usability. Thank you!

**Comment by JosiahWI** [comment]
[approve ci]

**Comment by JosiahWI** [comment]
PR title says "hinit", but it should say "hint".

File: cmake/AutoOptionHelpers.cmake [hunk 1]
```diff
@@ -21,17 +21,20 @@ set(GLOBAL_AUTO_OPTION_VARS "")
 
 function(_REGISTER_AUTO_OPTION _NAME _FEATURE_VAR _DESCRIPTION _DEFAULT)
   add_custom_target(${_NAME}_target)
-  set_target_properties(${_NAME}_target
-    PROPERTIES
-      AUTO_OPTION_FEATURE_VAR ${_FEATURE_VAR}
-  )
+  set_target_properties(${_NAME}_target PROPERTIES AUTO_OPTION_FEATURE_VAR ${_FEATURE_VAR})
```
Comment (SolidWallOfCode) [inline cmake/AutoOptionHelpers.cmake:24]: What's with the formatting? This was changed to be on one line, while line 29 was changed to be split.

**Comment by SolidWallOfCode** [comment]
Most of this PR is formatting changes - could you undo those to make the actual change clearer?

**Comment by JosiahWI** [comment]
The formatting changes are done by the formatter we agreed to use for consistency. Maybe it can be split into 2 separate commits?

File: cmake/AutoOptionHelpers.cmake (hunk 1 above)
Comment (cmcfarlen) [reply cmake/AutoOptionHelpers.cmake:24]: The cmake formatter is a fickle beast

Comment (JosiahWI) [reply cmake/AutoOptionHelpers.cmake:24]: I thought about this a little more. If the formatter is not run from the project root, it presumably will not detect our config and will use its defaults. We need a way to be confident this is actually our config and not the default one. I think we ran the formatter across the entire repository when we first started using it, so this module should **not** need to be reformatted now.

Comment (JosiahWI) [reply cmake/AutoOptionHelpers.cmake:24]: Having a pre-commit hook and a CMake target like we do for clang-format would help with this.

**Review by SolidWallOfCode — APPROVED** [review]

**Pull request merged**